components/ws-daemon/cmd/run.go

```diff
@@ -43,85 +38,34 @@ var runCmd = &cobra.Command{
 		if err != nil {
 			log.WithError(err).Fatal("cannot read configuration. Maybe missing --config?")
 		}
+
 		reg := prometheus.NewRegistry()
+		reg.MustRegister(
+			collectors.NewGoCollector(),
+			collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+		)
+
 		dmn, err := daemon.NewDaemon(cfg.Daemon, prometheus.WrapRegistererWithPrefix("gitpod_ws_daemon_", reg))
 		if err != nil {
 			log.WithError(err).Fatal("cannot create daemon")
 		}
 
-		common_grpc.SetupLogging()
-
-		grpcMetrics := grpc_prometheus.NewServerMetrics()
-		grpcMetrics.EnableHandlingTimeHistogram()
-		reg.MustRegister(grpcMetrics)
-
-		grpcOpts := common_grpc.ServerOptionsWithInterceptors(
-			[]grpc.StreamServerInterceptor{grpcMetrics.StreamServerInterceptor()},
-			[]grpc.UnaryServerInterceptor{grpcMetrics.UnaryServerInterceptor()},
+		health := healthcheck.NewHandler()
+		srv, err := baseserver.New(grpcServerName,
+			baseserver.WithGRPC(&cfg.Service),
+			baseserver.WithMetricsRegistry(reg),
+			baseserver.WithHealthHandler(health),
+			baseserver.WithGRPCHealthService(grpc_health.NewServer()),
```

Comment:
There's already a default server exposed in baseserver - https://github.com/gitpod-io/gitpod/blob/main/components/common-go/baseserver/options.go#L148